packages/design-system/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -29,7 +28,10 @@ import uniqueId from 'lodash.uniqueid';
  * @return {Boolean} Is this a TextField component?
  */
 function isTextField(child) {
-  return child != null && child.type === TextField;
+  return (
+    // Some application does not support child.type
+    child && child.type && child.type.displayName === 'TextField'
```

Comment:
We want to also use `type.name` as a backup, and I recommend using lodash to simplify this logic
```
import { get, uniqueId } from 'lodash';
return get(child, 'type.name') || get(child, 'type.displayName')
```